src/Grid/grid.jl

```diff
@@ -371,7 +371,12 @@ function getneighborhood(top::ExclusiveTopology, grid::AbstractGrid, vertexidx::
     cell_vertices = vertices(getcells(grid,cellid))
     global_vertexid = cell_vertices[local_vertexid]
     if include_self
-        return [top.vertex_vertex_neighbor[global_vertexid].neighbor_info; vertexidx]
+        self_reference_local = VertexIndex[]
+        for cellid in top.vertex_to_cell[global_vertexid]
+            local_vertex = VertexIndex(cellid,findfirst(x->x==global_vertexid,vertices(getcells(grid,cellid))))
+            push!(self_reference_local,local_vertex)
```

Comment:
Is there a performance hit in not preallocating the size? (we should have the size in the tables)